actors/miner/src/lib.rs

```diff
@@ -1616,26 +1616,27 @@ impl Actor {
                     return Err(actor_error!(illegal_argument, "too many deals for sector {} > {}", precommit.deal_ids.len(), deal_count_max));
                 }
 
-                let computed_cid = &verify_return.unsealed_cids[i];
-
                 // 1. verify that precommit.unsealed_cid is correct
                 // 2. create a new on_chain_precommit
 
-                let commd = precommit.unsealed_cid.unwrap();
-                if commd.0 != *computed_cid {
-                    return Err(actor_error!(illegal_argument, "computed {:?} and passed {:?} CommDs not equal",
-                            computed_cid, commd));
+                let declared_commd = precommit.unsealed_cid.unwrap();
```

Comment:
Tests were failing here because we were interpreting computed cid as compact commd and so no precommits with no deals and non zero CommD could pass this validation